Index: src/filemanager/cmd.c
===================================================================
--- a/src/filemanager/cmd.c	(revision 5c252726f1b52863a75b0118603a0e053923eded)
+++ b/src/filemanager/cmd.c	(revision 83ee61254bfbd653f1f47665749bb728b56e4593)
@@ -996,11 +996,11 @@
 reread_cmd (void)
 {
-    panel_update_flags_t flag = UP_OPTIMIZE;
-
-    if (!current_panel->is_panelized && get_current_type () == view_listing &&
-        get_other_type () == view_listing  && strcmp (current_panel->cwd, other_panel->cwd) == 0)
-        flag |= UP_RELOAD | UP_ONLY_CURRENT;
-
-    update_panels (flag, UP_KEEPSEL);
+    panel_update_flags_t flag = UP_ONLY_CURRENT;
+
+    if (get_current_type () == view_listing && get_other_type () == view_listing
+        && strcmp (current_panel->cwd, other_panel->cwd) == 0)
+        flag = UP_OPTIMIZE;
+
+    update_panels (UP_RELOAD | flag, UP_KEEPSEL);
     repaint_screen ();
 }
Index: src/filemanager/panel.c
===================================================================
--- a/src/filemanager/panel.c	(revision ac23f3277ba66274b8722af6ca4dce68318891e5)
+++ b/src/filemanager/panel.c	(revision 83ee61254bfbd653f1f47665749bb728b56e4593)
@@ -3438,5 +3438,8 @@
     {
         WPanel *panel;
+
         panel = (WPanel *) get_panel_widget (which);
+        if (panel->is_panelized)
+            flags &= ~UP_RELOAD;
         update_one_panel_widget (panel, flags, current_file);
     }
@@ -4157,5 +4160,6 @@
         panel = (WPanel *) get_panel_widget (get_other_index ());
 
-    ret = mc_chdir (panel->cwd);
+    if (!panel->is_panelized)
+        ret = mc_chdir (panel->cwd);
 }
 
