Index: lib/widget/input.c
===================================================================
--- orig/lib/widget/input.c	(revision 0138645541375862ac35bb525af55b46c0564110)
+++ new/lib/widget/input.c	(revision e1299dab3e8be829301142facc14ade0599f2425)
@@ -1277,11 +1277,11 @@
     {
         cp = str_term_substring (in->buffer, in->term_first_shown, in->field_width - has_history);
+        tty_setcolor (in->color[WINPUTC_MAIN]);
         for (i = 0; i < in->field_width - has_history; i++)
         {
-            if (i >= 0)
-            {
-                tty_setcolor (in->color[WINPUTC_MAIN]);
-                tty_print_char ((cp[0] != '\0') ? '*' : ' ');
-            }
+            if (i < (buf_len - in->term_first_shown) && cp[0] != '\0')
+                tty_print_char ('*');
+            else
+                tty_print_char (' ');
             if (cp[0] != '\0')
                 str_cnext_char (&cp);
