Index: lib/global.c
===================================================================
--- orig/lib/global.c	(revision 41af257135b103f31a3415c775421d9e907bf782)
+++ new/lib/global.c	(revision 2f03caefedef7726ffe43f1845d6af3718d45e90)
@@ -99,4 +99,5 @@
         .command_line_colors = NULL,
         .xterm_flag = FALSE,
+        .disable_x11 = FALSE,
         .slow_terminal = FALSE,
         .disable_colors = FALSE,
Index: lib/global.h
===================================================================
--- orig/lib/global.h	(revision 3e1b644f41ed6946a0201cdc299a6a170f32c19e)
+++ new/lib/global.h	(revision 2f03caefedef7726ffe43f1845d6af3718d45e90)
@@ -248,4 +248,7 @@
         gboolean xterm_flag;
 
+        /* disable x11 support */
+        gboolean disable_x11;
+
         /* For slow terminals */
         /* If true lines are shown by spaces */
Index: lib/tty/key.c
===================================================================
--- orig/lib/tty/key.c	(revision 0138645541375862ac35bb525af55b46c0564110)
+++ new/lib/tty/key.c	(revision 2f03caefedef7726ffe43f1845d6af3718d45e90)
@@ -668,5 +668,5 @@
 {
 #ifdef HAVE_TEXTMODE_X11_SUPPORT
-    if (getenv ("DISPLAY") != NULL)
+    if (getenv ("DISPLAY") != NULL && !mc_global.tty.disable_x11)
     {
         x11_display = mc_XOpenDisplay (0);
Index: src/args.c
===================================================================
--- orig/src/args.c	(revision 0138645541375862ac35bb525af55b46c0564110)
+++ new/src/args.c	(revision 2f03caefedef7726ffe43f1845d6af3718d45e90)
@@ -200,4 +200,11 @@
      &mc_args__force_xterm,
      N_("Forces xterm features"),
+     NULL
+    },
+
+    {
+     "no-x11", 'X', ARGS_TERM_OPTIONS, G_OPTION_ARG_NONE,
+     &mc_global.tty.disable_x11,
+     N_("Disable X11 support"),
      NULL
     },
