diff -urNp mc.orig/doc/man/mc.1.in mc/doc/man/mc.1.in
--- mc.orig/doc/man/mc.1.in	2011-09-20 03:00:00.000000000 +0300
+++ mc/doc/man/mc.1.in	2011-10-07 17:30:33.000000000 +0300
@@ -3532,8 +3532,8 @@ with the assignment of colors, as descri
 Colors\&.
 .\"Colors"
 .PP
-If your skin contains any of 256\-color definitions, you should define the
-'256colors' key set to TRUE value in [skin] section.
+If your skin contains any of 256\-color definitions, you should define
+the '256colors' key set to TRUE value in [skin] section.
 
 .PP
 A skin\-file is searched on the following algorithm (to the first one found):
diff -urNp mc.orig/doc/man/ru/mc.1.in mc/doc/man/ru/mc.1.in
--- mc.orig/doc/man/ru/mc.1.in	2011-09-20 03:00:00.000000000 +0300
+++ mc/doc/man/ru/mc.1.in	2011-10-07 17:32:50.000000000 +0300
@@ -1494,7 +1494,7 @@ искать в каталогах, по
 Компоненты списка должны быть разделены двоеточиями. Например
 .PP
 .nf
-.cdrom:/nfs/wuarchive:/afs
+/cdrom:/nfs/wuarchive:/afs
 .fi
 .PP
 Относительные пути также поддерживаются. Следующий пример показывает, как
